[이다빈] Mission 4: 로그인 기능 구현(1차 수정)

thisisempty opened the pull request:
>wecode PR message template 및 체크 리스트 입니다. 
아래 사항들을 전부 작성/체크 하시고 PR 해주세요!

## 수정 사항 간략한 한줄 요약
(여기에 수정 사항에 대한 간략한 한줄 요약/제목 작성해 주세요.)
- confilct 해결
- 코딩 컨벤션에 맞추어 수정
- 사용하지 않는 모듈 제거
		 
## 수정 사항들 자세한 내용
(여기에 수정 사항에 대한 자세한 내용을 작성해 주세요.)
- 이메일과 패스워드로 로그인 할 수 있음
  - 이메일과 패스워드가 전달되지 않으면 400 에러 반환
  - 이메일이나 비밀번호가 일치하지 않으면 401 에러 반환
  - 성공시 http 200 status 반환

## 기타 질문 및 특이 사항
(궁금한 사항들, 하면서 느낀 점들 공유해주실 것이 있으면 자유롭게 작성해 주세요.)
 

## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
조건을 만족한다면 각 빈칸에 x 표시 해주세요
- [x] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [x] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [x] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [x] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.

thisisempty committed:
[Modify]
Change the phone column from an IntegerField to a CharField.

kyeongraekim commented on students/dabeen/westagram/users/views.py:
```diff
@@ -1,3 +1,30 @@
+    def get(self, request):
+        data = json.loads(request.body)
+        try:
+
```
try문 시작 직후 불필요한 라인 공백을 삭제해주세요.

kyeongraekim commented on students/dabeen/westagram/users/views.py:
```diff
@@ -1,3 +1,30 @@
+
+                if user.password == data['password']:
+                    return JsonResponse({"message": "SUCCESS"}, status=200)
+                # If user enter the wrong password return 401 ERROR
```
코드 리딩으로 충분히 이해 가능한 부분으로서, 주석으로의 별도 설명이 불필요해 보입니다.

해당 라인 이하 주석들도 동일하게 적용해서 삭제해주세요~

kyeongraekim commented on students/dabeen/westagram/users/views.py:
```diff
@@ -1,3 +1,30 @@
 from django.shortcuts import render
```
사용하지 않는 import문은 삭제해 주세요.

kyeongraekim commented on students/dabeen/westagram/users/views.py:
```diff
@@ -1,3 +1,30 @@
+            if User.objects.filter(email=data['email']).exists():
+                user = User.objects.get(email=data['email'])
+
+                if user.password == data['password']:
```
논리 구조상 if문 내부에 또다른 if문이 중첩되어서 들어가는 구조는 불필요해 보입니다. if 문 사이의 indentation 관계를 다시 한 번 생각해보세요.

kyeongraekim reviewed (changes requested):
안녕하세요 다빈님~ 지난 회원가입 기능 구현 이후 수정 시 같이 수정해주세요~
앞에 PR이 머지되면 해당 이번 로그인 기능 PR은  conflict가 날 가능성이 높습니다.
이후에 직접 컨플릭트도 해결해보는 시간을 가져보면 좋을 것 같습니다.

kyeongraekim commented on students/dabeen/westagram/users/views.py:
```diff
@@ -1,3 +1,30 @@
+                user = User.objects.get(email=data['email'])
+
+                if user.password == data['password']:
+                    return JsonResponse({"message": "SUCCESS"}, status=200)
```
로그인 성공의 로직을 중간에 두는 것보다 모든 검사를 끝내고 마지막에 로그인 성공!의 논리전개로 변경하시면 가독성과 코드 효율성이 더 올라 갈 것 같습니다.

thisisempty committed:
# This is a combination of 3 commits.
# This is the 1st commit message:

modify phone columm max_lengt 20 to 100

# This is the commit message #2:

[ADD]
users/urls.py : add SignInView path

[Modify]
users/views.py : delete duplicated part

# This is the commit message #3:

modify duplicated part

kyeongraekim commented:
다빈님~ 수정 내용 잘 확인했습니다. 다만 아직 컨플릭트가 해결이 안되있는 것 같아요. 확인 후 수정부탁드립니다.
혹시 컨플릭트 해결이 힘드시면 언제든 라운지에 나와서 문의 주세요!

thisisempty committed:
resolve conflict_rebase_trial

thisisempty committed:
resolve conflict_4th_trial

thisisempty committed:
modify phone columm max_lengt 20 to 100

[ADD]
users/urls.py : add SignInView path

[Modify]
users/views.py : delete duplicated part

modify duplicated part

[ADD]
users/urls.py : new signin url

[CREATE]
westagram/views.py : new SignInView

[Modify]
Change the phone column from an IntegerField to a CharField.

modify phone columm max_lengt 20 to 100

remove unused modules

resolve conflict

thisisempty committed:
urls.py modified

thisisempty committed:
views.py modified

kyeongraekim commented:
새 브랜치 작업으로 인한 PR Closing 입니다.
